Update StringParsingBuffer to use std::span

cdumez opened the pull request:
#### d2a029f757b316ef28a6935af9d87a3b9d9d72e2
<pre>
Update StringParsingBuffer to use std::span
<a href="https://bugs.webkit.org/show_bug.cgi?id=272630">https://bugs.webkit.org/show_bug.cgi?id=272630</a>

Reviewed by Darin Adler.

Update StringParsingBuffer to use std::span.

* Source/JavaScriptCore/runtime/ISO8601.cpp:
(JSC::ISO8601::parseTimeSpec):
* Source/WTF/wtf/text/StringParsingBuffer.h:
(WTF::readCharactersForParsing):
* Source/WebCore/css/parser/CSSCustomPropertySyntax.cpp:
(WebCore::CSSCustomPropertySyntax::parseComponent):
(WebCore::CSSCustomPropertySyntax::parse):
* Source/WebCore/css/parser/CSSCustomPropertySyntax.h:
* Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
(WebCore::HTMLFastPathParser::HTMLFastPathParser):
* Source/WebCore/loader/appcache/ApplicationCacheManifestParser.cpp:
(WebCore::parseApplicationCacheManifest):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
(WebCore::ContentSecurityPolicyDirectiveList::parse):
(WebCore::ContentSecurityPolicyDirectiveList::parseDirective):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.h:
* Source/WebCore/page/csp/ContentSecurityPolicySourceList.cpp:
(WebCore::ContentSecurityPolicySourceList::parse):
(WebCore::ContentSecurityPolicySourceList::parseSource):
(WebCore::ContentSecurityPolicySourceList::parseHost):
(WebCore::ContentSecurityPolicySourceList::parsePath):
(WebCore::ContentSecurityPolicySourceList::parsePort):
* Source/WebCore/page/csp/ContentSecurityPolicySourceList.h:
* Source/WebCore/page/csp/ContentSecurityPolicyTrustedTypesDirective.cpp:
(WebCore::ContentSecurityPolicyTrustedTypesDirective::parse):
* Source/WebCore/svg/SVGParserUtilities.cpp:
(WebCore::parseUnicodeRange):
(WebCore::parseKerningUnicodeString):
* Source/WebCore/svg/SVGPathStringViewSource.cpp:
(WebCore::SVGPathStringViewSource::SVGPathStringViewSource):
* Tools/TestWebKitAPI/Tests/WTF/StringParsingBuffer.cpp:
(TestWebKitAPI::TEST(WTF, StringParsingBufferInitial)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferAdvance)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferAdvanceBy)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferPreIncrement)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferPostIncrement)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferPlusEquals)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferEnd)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferSubscript)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferStringView)):

Canonical link: <a href="https://commits.webkit.org/277466@main">https://commits.webkit.org/277466@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/176967149a4c23b4a42e4f9da8e52c51000b2b52

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47681 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26869 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50426 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50364 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43736 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/49988 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/32721 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24346 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38815 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48263 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24528 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/41150 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20112 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22009 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/42342 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5730 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/40963 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44030 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/42763 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52256 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/47172 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22717 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19051 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/46119 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23989 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45150 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24777 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/54672 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6739 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23709 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11215 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/JavaScriptCore/runtime/ISO8601.cpp:
```diff
@@ -348,8 +348,8 @@ static std::optional<PlainTime> parseTimeSpec(StringParsingBuffer<CharacterType>
-    unsigned digits = 0;
-    unsigned maxCount = std::min(buffer.lengthRemaining(), 9u);
+    size_t digits = 0;
+    size_t maxCount = std::min<size_t>(buffer.lengthRemaining(), 9u);
```
Don’t need both the `u` and the `size_t`.

cdumez replied:
```
./runtime/ISO8601.cpp:352:23: error: no matching function for call to 'min'
  352 |     size_t maxCount = std::min(buffer.lengthRemaining(), 9);
```

darinadler replied:
I meant that if you need and use `<size_t>` you don’t **also** need the `u`.

cdumez replied:
Oh, my bad. The good news is that I ended up committing the right thing.

cdumez committed:
Update StringParsingBuffer to use std::span
https://bugs.webkit.org/show_bug.cgi?id=272630

Reviewed by Darin Adler.

Update StringParsingBuffer to use std::span.

* Source/JavaScriptCore/runtime/ISO8601.cpp:
(JSC::ISO8601::parseTimeSpec):
* Source/WTF/wtf/text/StringParsingBuffer.h:
(WTF::readCharactersForParsing):
* Source/WebCore/css/parser/CSSCustomPropertySyntax.cpp:
(WebCore::CSSCustomPropertySyntax::parseComponent):
(WebCore::CSSCustomPropertySyntax::parse):
* Source/WebCore/css/parser/CSSCustomPropertySyntax.h:
* Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
(WebCore::HTMLFastPathParser::HTMLFastPathParser):
* Source/WebCore/loader/appcache/ApplicationCacheManifestParser.cpp:
(WebCore::parseApplicationCacheManifest):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
(WebCore::ContentSecurityPolicyDirectiveList::parse):
(WebCore::ContentSecurityPolicyDirectiveList::parseDirective):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.h:
* Source/WebCore/page/csp/ContentSecurityPolicySourceList.cpp:
(WebCore::ContentSecurityPolicySourceList::parse):
(WebCore::ContentSecurityPolicySourceList::parseSource):
(WebCore::ContentSecurityPolicySourceList::parseHost):
(WebCore::ContentSecurityPolicySourceList::parsePath):
(WebCore::ContentSecurityPolicySourceList::parsePort):
* Source/WebCore/page/csp/ContentSecurityPolicySourceList.h:
* Source/WebCore/page/csp/ContentSecurityPolicyTrustedTypesDirective.cpp:
(WebCore::ContentSecurityPolicyTrustedTypesDirective::parse):
* Source/WebCore/svg/SVGParserUtilities.cpp:
(WebCore::parseUnicodeRange):
(WebCore::parseKerningUnicodeString):
* Source/WebCore/svg/SVGPathStringViewSource.cpp:
(WebCore::SVGPathStringViewSource::SVGPathStringViewSource):
* Tools/TestWebKitAPI/Tests/WTF/StringParsingBuffer.cpp:
(TestWebKitAPI::TEST(WTF, StringParsingBufferInitial)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferAdvance)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferAdvanceBy)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferPreIncrement)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferPostIncrement)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferPlusEquals)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferEnd)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferSubscript)):
(TestWebKitAPI::TEST(WTF, StringParsingBufferStringView)):

Canonical link: https://commits.webkit.org/277466@main
